hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterMergeTask.java

```diff
@@ -67,7 +71,7 @@ public void run() {
             Collection<Runnable> coreTasks = collectMergeTasks(true);
             Collection<Runnable> nonCoreTasks = collectMergeTasks(false);
 
-            resetServices();
+            resetServices(true);
```

Comment:
I think we can have 2 separate methods for `resetServices()`, one for the core services and another for the others

Reply:
agree 👍

Reply:
fixed